classes/metric/base.php

```diff
@@ -125,6 +125,29 @@ public function get_settings_url() {
         return null;
     }
 
+    /**
+     * The range for which a backfilled metric has retrieved data.
+     *
+     * @return array|bool First value is min timestamp, second is max timestamp, third is used interval.
+     * @throws \dml_exception
+     */
+    public function get_range_retrieved() {
+        $config = explode('-', get_config('tool_cloudmetrics', $this->get_name() . '_range'));
+        if (count($config) !== 3) {
+            return [-1, -1, -1];
+        }
+        return [$config[0], $config[1], $config[2]];
+    }
+
+    /**
+     * Sets the range for which a backfilled metric has retrieved data.
+     *
+     * @param string $currentconfig Config containing min and max timestamps plus interval.
+     */
+    public function set_range_retrieved(string $currentconfig) {
```

Comment:
would it now make sense for set and get to take the same shaped data,ie array of 3 things

Reply:
Fixed.